git-hooks/pre-push

```diff
@@ -1,50 +1,68 @@
-#!/bin/sh
-#
-# Pre-push hooks
+#!/usr/bin/env node
+/* eslint-disable no-console */
 
-branch=$(git symbolic-ref --short HEAD)
+const path = require('path');
+const execa = require('execa');
+const chalk = require('chalk');
 
+const diff = () =>
+    execa.stdout('git', [
+        'diff',
+        '--name-only',
+        '--cached',
+        '--diff-filter=AM',
+    ]);
 
-conditionallyValidate() {
+const validate = () =>
+    execa
+        .stdout('make', ['validate-head'], { stdio: 'inherit' })
```

Comment:
do you think `make validate-head` is ever likely to be run outside of this context? does it need to be a make task do you think?